### DISCUSSION TITLE
Folders in a bucket can be shared, yet using the link results in 'xml error response'

### DISCUSSION THREAD

**Discussion by karl-dv** [minio/minio#18528] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
It's not a pleasant feeling when a customer complains about having received a broken link, just because a user forgot that in MinIO folders cannot be shared, although the MinIO Console user interface does allow to generate a normal looking 'share link'.

**Describe the solution you'd like**
Ideally the server should be able to handle the download request, i.e. by building a ZIP file with the last version of every object that resides in the shared folder (to which the generated 'share link' points to), and sending that ZIP file as a response for the download request. Or for a versioned bucket, a ZIP file based on the objects versions as they were at the time that the link was generated.
PS: Technically a ZIP can be generated on the fly, as a byte-stream that can be piped directly to the web server response, thereby completely eliminating the struggle of having to save it as a temp file on the server and cleaning it up after sending the response.

**Describe alternatives you've considered**
On the other hand, if (or while) the server is not able to generate a ZIP file for 'share links' that point to folders, the MinIO Console should prevent the user from generating a 'share link' for folders, or at least issue a warning or so. That would certainly not be the ideal solution, but at least it would prevent the users of generating links that turn out to be rather useless.

**Additional context**
When in MinIO Console, the user can select a folder and click on 'Share' to generate a link, just like when selecting a 'file' object.
However, when using the generated 'share link' inside a browser (or inside a CURL session or other), the server sends the following response:
`<Error>`
`<Code>NoSuchVersion</Code>`
`<Message>The specified version does not exist.</Message>`
`<Key>test/</Key>`
`<BucketName>example-bucket</BucketName>`
`<Resource>/example-bucket/test/</Resource>`
`<RequestId>179A41A79C0D8558</RequestId>`
`<HostId>dd9125bab4ad424b049176c95eb6ebf372d3b3fd1af9251948b608df7dc2b4e5</HostId>`
`</Error>`
Looking at the message, I suspect that this options probably works fine it the bucket is not versioned. But since we have setup our server to be replicated to another server, I am unable to create an unversioned bucket for testing that scenario.

**Comment by klauspost** [comment, marked as answer]
There are no "share folder" or "download as zip" functionality in the S3 API. 

Prefixes shouldn't really have a "Share" option. We will disable that for an upcoming release.
